File: src/main/java/org/usfirst/irs1318/gamesim/engine/actions/ChangePhase.java
package org.usfirst.irs1318.gamesim.engine.actions;

import org.usfirst.irs1318.gamesim.engine.Action;
import org.usfirst.irs1318.gamesim.engine.Field;
import org.usfirst.irs1318.gamesim.engine.TaskDetails;

public final class ChangePhase extends Action {
    private final Field.Phase phase;

    public ChangePhase(Field actor, TaskDetails taskDetails, Field.Phase phase) {
        super(actor, taskDetails);
        this.phase = phase;
    }

    @Override
    public void performSuccessAction() {
        ((Field) actor).setPhase(phase);
        // TODO
    }

    @Override
    public void performFailureAction() {
        // TODO
    }

    public Field.Phase getPhase() {
        return phase;
    }
}

File: src/main/scala/org/usfirst/irs1318/gamesim/engine/action/ChangePhase.scala
package org.usfirst.irs1318.gamesim.engine.action

import org.usfirst.irs1318.gamesim.engine.GameEngine

case class ChangePhase(phase: GameEngine.State.Phase) extends Action {
  override def run(state: GameEngine.State): GameEngine.State = state.mapPhase(_ => phase)
}
